load/scan.go

```diff
@@ -93,82 +95,108 @@ func scanWithIndexer(channels []*duplexChannel, batchSize uint, limit uint64, br
+	// 2. outstandingBatches, which contains batches, ready to be sent to worker.
+	//    As soon as there is place in chan to worker, batch is placed into chan
+
+	// Current batches (per channel) that are being filled with items from scanner
+	fillingBatches := make([]Batch, numChannels)
+	for i := range fillingBatches {
+		fillingBatches[i] = factory.New()
 	}
 
 	// Batches that are ready to be set when space on a channel opens
-	unsent := make([][]Batch, numChannels)
-	for i := range unsent {
-		unsent[i] = []Batch{}
+	outstandingBatches := make([][]Batch, numChannels)
+	for i := range outstandingBatches {
+		outstandingBatches[i] = []Batch{}
 	}
 
 	// We use Select via reflection to either select an acknowledged channel so
 	// that we can potentially send another batch, or if none are ready to continue
 	// on scanning. However, when we reach a limit of outstanding batches, we also
 	// want to block until one worker is done, so as not to starve the workers.
-	// Using an array with Select via reflection gives us this flexibility (i.e., we can
-	// either pass the whole array of cases, or the array less the last item).
+	// Using an array with Select via reflection gives us this flexibility (i.e.,
+	// we can either pass the whole array of cases, or the array less the last item).
 	cases := make([]reflect.SelectCase, numChannels+1)
 	for i, ch := range channels {
-		cases[i] = reflect.SelectCase{Dir: reflect.SelectRecv, Chan: reflect.ValueOf(ch.toScanner)}
+		cases[i] = reflect.SelectCase{
+			Dir:  reflect.SelectRecv,
+			Chan: reflect.ValueOf(ch.toScanner),
+		}
+	}
+	cases[numChannels] = reflect.SelectCase{
+		Dir: reflect.SelectDefault,
 	}
-	cases[numChannels] = reflect.SelectCase{Dir: reflect.SelectDefault}
 
-	// Keep track of how many batches are outstanding (ocnt) so we don't go over
-	// a limit (olimit), in order to slow down the scanner so it doesn't starve the workers
+	// Keep track of how many batches are outstanding (ocnt),
+	// so we don't go over a limit (olimit), in order to slow down the scanner so it doesn't starve the workers
 	ocnt := 0
 	olimit := numChannels * cap(channels[0].toWorker) * 3
 	for {
+
+		// Check whether incoming items limit reached.
+		// We do not want to process more items than specified.
 		if limit > 0 && itemsRead == limit {
 			break
 		}
 
 		caseLimit := len(cases)
-		if ocnt >= olimit { // if we have too many outstanding, wait until one finishes (i.e. no default)
+		if ocnt >= olimit {
+			// We have too many outstanding batches, wait until one finishes (i.e. no default)
 			caseLimit--
 		}
 
 		// Only receive an 'ok' when it's from a channel, default does not return 'ok'
 		chosen, _, ok := reflect.Select(cases[:caseLimit])
 		if ok {
-			unsent[chosen] = ackAndMaybeSend(channels[chosen], &ocnt, unsent[chosen])
+			outstandingBatches[chosen] = sendBatch(channels[chosen], &ocnt, outstandingBatches[chosen])
```

Comment:
Still prefer `unsent`. Perhaps `unsentBatches`.

Reply:
Ok, let it be `unsentBatches` then. Still seems useful to add more clarity to var names, so `unsentBatches` looks like clearer to me than simple `unsent`.
Updated accordingly.